### PR TITLE
Guard MEFISTO code with version check against SMESH.

### LINKED ISSUE #11746: [Problem/Build Error] Remove MEFISTO components
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

At least in Salome 9.10 the header for integrating MEFISTO is completly gone. So you will be unable to build FreeCAD with a newer version of Salome. 

### Full version info

```shell
Tag 0.21.2
```

### Subproject(s) affected?

FEM

### Anything else?

Removing commit: https://git.salome-platform.org/gitweb/?p=modules/smesh.git;a=commit;h=9d7121c884b58f35182dab517602612c1606a6bf

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Neumann-A**
closes #11746

### CODE REVIEW

**Comment by adrianinsaval** [comment]
Anybody knows what is mefisto used for?

**Comment by WandererFan** [comment]
> Anybody knows what is mefisto used for?
"solve some Partial Differential Equations by Finite Element methods"

File: src/Mod/Fem/App/HypothesisPy.cpp [hunk 1]
```diff
@@ -1178,7 +1180,7 @@ StdMeshers_MEFISTO_2DPy::StdMeshers_MEFISTO_2DPy(int hypId, int studyId, SMESH_G
     : SMESH_HypothesisPyBase(new StdMeshers_MEFISTO_2D(hypId, studyId, gen))
 {}
 #endif
-
+#endif
 StdMeshers_MEFISTO_2DPy::~StdMeshers_MEFISTO_2DPy() = default;
```
Comment (Neumann-A) [inline src/Mod/Fem/App/HypothesisPy.cpp:1185]: This endif is misplaced and needs to be one line below.

Comment (chennes) [reply src/Mod/Fem/App/HypothesisPy.cpp:1185]: Are you submitting a PR to adjust?

Comment (Neumann-A) [reply src/Mod/Fem/App/HypothesisPy.cpp:1185]: #12268

**Pull request merged**